app/services/podcasts/get_episode.rb

```diff
@@ -0,0 +1,30 @@
+module Podcasts
+  class GetEpisode
+    def initialize(podcast)
+      @podcast = podcast
+    end
+
+    def call(item)
+      ep = existing_episode(item, podcast).first
+      if ep
+        Podcasts::UpdateEpisode.call(ep, item)
+      else
+        Podcasts::CreateEpisode.call(podcast.id, item)
+      end
+    end
+
+    private
+
+    attr_reader :podcast
+
+    # returns empty array if an episode doesn't exist
+    def existing_episode(item, podcast)
```

Comment:
I think this can be refactored with something like:

```ruby
PodcastEpisode.where(media_url: "url").or(PodcastEpisode.where(title: "title")).or(PodcastEpisode.where(guid: "guid")).exists? || (podcast.unique_website_url? && ....)
```

Currently in the worst case scenario you do 4 queries, with the refactoring you do 2 at worst

Reply:
I haven't touched this method yet to make the changes step by step. I'll use your notes when implementing further changes.